llvm/utils/TableGen/MacroFusionPredicatorEmitter.cpp

```diff
@@ -146,12 +148,9 @@ void MacroFusionPredicatorEmitter::emitFirstPredicate(Record *Predicate,
                  << (Predicate->getValueAsBit("ReturnValue") ? "true" : "false")
                  << ";\n";
   } else if (Predicate->isSubClassOf("OneUsePred")) {
-    OS.indent(2) << "{\n";
-    OS.indent(4) << "Register FirstDest = FirstMI->getOperand(0).getReg();\n";
-    OS.indent(4)
-        << "if (FirstDest.isVirtual() && !MRI.hasOneNonDBGUse(FirstDest))\n";
-    OS.indent(4) << "  return false;\n";
-    OS.indent(2) << "}\n";
+    OS.indent(2) << "if (!IsPostRA && "
```

Comment:
I don't understand why this is an improvement over checking isVirtual? Are we even guaranteed that a macro-fused instruction can't have a physical register in the pre-RA scheduler?

Reply:
You are right, I'm thoughtless about this. I'll revert this part.